Frogies-Project/Assets/Scripts/Core/ObjectPoolers/ObjectPooler.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Core.ObjectPoolers
+{
+    public class ObjectPooler
+    {
+        private Dictionary<string, Queue<GameObject>> _poolDictionary;
+
+        public static ObjectPooler Instance => _instance ??= new ObjectPooler();
+        private static ObjectPooler _instance;
+
+        [Serializable]
+        public class Pool
+        {
+            public string Tag { get; set; }
+            public GameObject Prefab { get; set; }
+            public int Size { get; set; }
+        }
+        
+        public void Init(List<Pool> pools)
```

Comment:
Imho, code duplication. It does the same thing as AddPool. Remove it and init dictionary in the constructor